package.json

```diff
@@ -84,13 +73,13 @@
   "jest": {
     "automock": false,
     "testEnvironment": "node",
-    "setupFiles": [
-      "./setupJest.js"
-    ],
+    "setupFiles": ["./setupJest.js"],
     "testPathIgnorePatterns": [
-      ".eslintrc.js",
       "<rootDir>/build/",
       "<rootDir>/node_modules/"
     ]
+  },
+  "prettier": {
+    "semi": false
```

Comment:
semicolons are important, please change this to true.

I also would like to use single quotes, which I think is controlled by `"singleQuote": true` https://prettier.io/docs/en/options#quotes

These two changes should also help reduce the amount of diff churn on the run prettier commit.

Reply:
I've switched it to use single quotes, and as you pointed out the diff is much smaller now. Makes it nicer to review for sure.

## As for semicolons

I'm new here, so we will go in whatever direction you say. But if I may have a few minutes of your time, I'd like to argue my point:

They are important where necessary, and useless everywhere else. It comes down to personal preference. I prefer not to see useless things, but I can endure.

- Prettier never changes how the code _works_
- Prettier only removes them where they are unnecessary (unimportant)
- Prettier does not remove them where they serve a purpose
- It will still add semis where there are common pitfalls for humans
- It will wrap lines where there might be ambiguity, making it obvious to the human how that code will run
- You can still write semicolons as you code, if you wish, then let prettier do its thing
- If we have semis ON, I can still choose to code without them, then let prettier do its thing

### Example 1

```
a = ["a"]
(a + b)
```

becomes...

```
a = ["a"](a + b)
```

_Uh oh, that turned into a function call, I see what I made wrong there..._

### Example 2

```
a = ["a"];
(a + b) * 3;
```

becomes...

```
a = ["a"]
;(a + b) * 3
```

_What? A semicolon was inserted before my row? Maybe I did something odd here..._

To add some weight to my argument; Since I changed my mind about them back in 2017 I've had some 30 people working without semicolons and we've never had a bug resulting from omitting them while Prettier does its magic. I can count on one hand where I made an assumption about ASI that proved wrong, and in each case it was obvious.

Reply:
Thank you for the explanation. I've given this a lot of thought over the weekend.

As you pointed out, we can choose to type the semicolon or not type it, and prettier will add it or remove it based on the configuration, which is very nice.

While I understand your point about not wanting to see things that are useless, I'd like to offer the following thoughts:

- I find the semicolons to make the code easier to read. I think of it as punctuation at the end of a sentence, which helps break things up for my eyes. Agreed that this is a personal preference and that others may find the opposite to be true.
- You mentioned that there are a few cases where semicolons are required in order for the code to be parsed and executed correctly. I tend to favor rules that are biased towards simplicity. In this case, since semicolons are sometimes required it is a simpler rule to say "semicolons should always be used" vs "semicolons should never be used except when they are required to avoid bugs". Though it is nice that the tooling can handle this for us and in practice we should not need to think about it, I still favor using the simpler rule.
- This project has historically required semicolons, and there is some value in maintaining that inertia and not making this change. Granted, that's not a great argument for making changes that are important, but as you said this comes down to personal preference and is not a critical change (vs. e.g. fixing a bug or adding a useful feature).
- By sticking with the existing format of requiring semicolons, we will reduce git diff churn. This will make reviewing this PR easier (minor benefit) and will reduce friction when using log diving tools like `git blame` (medium benefit).

At minimum, if you want to keep pursuing this, it would be nice to move the controversial bits to a separate PR so we can land the rest of this one and move forward with other work that builds on top of it.